js2-mode.el

```diff
@@ -10878,6 +10878,70 @@ And, if CHECK-ACTIVATION-P is non-nil, use the value of TOKEN."
         (js2-check-activation-name s (or token js2-NAME)))
     name))
 
+;;; Use AST to extract semantic information
+
+(defun js2-get-element-index-from-array-node (elem array-node &optional hardcoded-array-index)
+  "Get index of ELEM from ARRAY-NODE or 0 and return it as string."
+  (let ((idx 0) elems (rlt hardcoded-array-index))
+    (setq elems (js2-array-node-elems array-node))
+    (if (and elem (not hardcoded-array-index))
+        (setq rlt (catch 'nth-elt
+                    (dolist (x elems)
+                      ;; Is same node by comparing position?
+                      ;; Since we know the ELEM does belong to ARRAY-NODE,
+                      ;; we could check only offset field for performance reason.
+                      (if (= (js2-node-pos elem) (js2-node-pos x))
```

Comment:
which one is more efficient? Is eq more efficient than two integer operations (one integer plus plus one integer compare)?